lib/postgrex/protocol.ex

```diff
@@ -702,7 +712,43 @@ defmodule Postgrex.Protocol do
 
     digest = :crypto.hash(:md5, [pass, user]) |> Base.encode16(case: :lower)
     digest = :crypto.hash(:md5, [digest, salt]) |> Base.encode16(case: :lower)
-    auth_send(s, msg_password(pass: ["md5", digest]), status, buffer)
+    auth_send(s, msg_password(pass: ["md5", digest, 0]), status, buffer)
+  end
+
+  defp auth_sasl(s, status = _, buffer) do
+    nonce = Enum.join(
+      for _ <- 1..@nonce_length do
+        "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ+/"
+        |> String.codepoints
+        |> Enum.random
+      end
+    )
+    message = "n,,n=,r=" <> nonce
```

Comment:
We don't need to call `Enum.join` nor use binary concatenation, let's work with IO lists here.